### PR TITLE
Extend product quantities section

### PR DESCRIPTION
**Pull request by IKarbowiak**
Extend the `Stock quantities` section with the explanation of how the `quantityAvailable` is calculated.

### CODE REVIEW

**Review by fowczarek — APPROVED** [review]

File: docs/developer/products.mdx [hunk 1]
```diff
@@ -509,14 +513,94 @@ Response:
 }
 ```
 
-- `"quantityAvailable": 50`: customers only see maximum of 50 pcs available to buy
-- `stocks` contain full list of stock quantities in every of the warehouses
+- the `quantityAvailable` is a sum of stock quantities from all available stocks,
+  in this case, the sum is bigger than the `MAX_CHECKOUT_LINE_QUANTITY` so the maximum
+  value is returned - 50 pcs available to buy
```
Comment (peelar) [inline docs/developer/products.mdx:518]:
```suggestion
- The `quantityAvailable` is a sum of stock quantities from all available stocks.
  In this case, the sum is bigger than the `MAX_CHECKOUT_LINE_QUANTITY` so the maximum
  value of 50 products is returned.
```

File: docs/developer/products.mdx [hunk 2]
```diff
@@ -509,14 +513,94 @@ Response:
...
+- `stocks` contain a full list of stock quantities in every of the warehouses
```
Comment (peelar) [inline docs/developer/products.mdx:519]:
```suggestion
- `stocks` contain a full list of stock quantities in each warehouse.
```

File: docs/developer/products.mdx [hunk 3]
```diff
@@ -509,14 +513,94 @@ Response:
...
+
+:::note
+When the product has a track inventory turned off, the `quantityAvaible` will be always
+equal to `MAX_CHECKOUT_LINE_QUANTITY` value.
```
Comment (peelar) [inline docs/developer/products.mdx:523]:
```suggestion
When the product has a track inventory turned off, the `quantityAvailable` will always
equal to the `MAX_CHECKOUT_LINE_QUANTITY` value.
```

File: docs/developer/products.mdx [hunk 4]
```diff
@@ -509,14 +513,94 @@ Response:
...
+:::
+
+### Getting the available quantity for the particular address
+
+The quantity available can be fetched for the given address. The returned quantity
+is the sum of the quantity from all stocks that are available in the shipping zone that
+support provided address. When the address is not provided the highest quantity
```
Comment (pawelpikus) [inline docs/developer/products.mdx:530]:
```suggestion
supports provided address. When the address is not provided the highest quantity

```

File: docs/developer/products.mdx [hunk 5]
```diff
@@ -509,14 +513,94 @@ Response:
...
+          "quantityAllocated": 0
+        }
+      ]
+    }
+  }
+}
+```
+
+As a response we get the quantity 20 as the `americas` is the only warehouse that is
+available in the `US`.
```
Comment (pawelpikus) [inline docs/developer/products.mdx:582]:
```suggestion
In the response, we get the available quantity of 20 as the `americas` is the only warehouse available in the `US`.
````

File: docs/developer/products.mdx [hunk 6]
```diff
@@ -509,14 +513,94 @@ Response:
...
+
+In the case when the address is not provided, the quantity for the shipping zone with
+the highest available quantity is returned.
+
+### Handling collection point warehouses in the quantity calculations
+
+The collection point warehouses are treated as a magic single-warehouse shipping zone.
+
+- In the case of a local collection point warehouse, the calculated quantity will be equal
+  to the quantity of this local warehouse.
```
Comment (pawelpikus) [inline docs/developer/products.mdx:592]:
```suggestion
- In the case of a local collection point warehouse, the calculated quantity will equal the quantity of this local warehouse.
```

File: docs/developer/products.mdx [hunk 7]
```diff
@@ -509,14 +513,94 @@ Response:
...
+- In the case of a global collection point warehouse the quantity will be equal to the sum
```
Comment (pawelpikus) [inline docs/developer/products.mdx:593]:
```suggestion
- In the case of a global collection point warehouse the quantity will equal the sum
```

**Review by pawelpikus — CHANGES_REQUESTED** [review]
Hi,
Looking good overall. I just got rid of the passive when it was not necessary, and fixed some minor grammar mistakes.

Hope it helps!

**Review by peelar — APPROVED** [review]

File: docs/developer/products.mdx [hunk 8]
```diff
@@ -424,6 +426,7 @@ Result:
 The `address` parameter is used to check the quantity in a warehouse that is connected with a shipping zone that includes the given address. If not provided, Saleor will use a warehouse with the highest available quantity.
```
Comment (pawelpikus) [inline docs/developer/products.mdx:426]:
```suggestion
The `address` parameter checks the quantity in the warehouse connected with the shipping zone that includes the given address. If not provided, Saleor will use the warehouse with the highest available quantity.
```

File: docs/developer/products.mdx [hunk 9]
```diff
@@ -509,14 +513,94 @@ Response:
 }
 ```
 
-- `"quantityAvailable": 50`: customers only see maximum of 50 pcs available to buy
-- `stocks` contain full list of stock quantities in every of the warehouses
+- The `quantityAvailable` is a sum of stock quantities from all available stocks.
+  In this case, the sum is bigger than the `MAX_CHECKOUT_LINE_QUANTITY` so the maximum
+  value of 50 products is returned.
```
Comment (pawelpikus) [inline docs/developer/products.mdx:518]:
```suggestion
  In this case, the sum is higher than the `MAX_CHECKOUT_LINE_QUANTITY`, so the maximum value of 50 products is returned.
```

File: docs/developer/products.mdx [hunk 10]
```diff
@@ -509,14 +513,94 @@ Response:
...
+- `stocks` contain a full list of stock quantities in each warehouse.
+
+:::note
+When the product has a track inventory turned off, the `quantityAvailable` will be always
```
Comment (pawelpikus) [inline docs/developer/products.mdx:522]:
```suggestion
When the product has the track inventory turned off, the `quantityAvailable` will always be
```

File: docs/developer/products.mdx [hunk 11]
```diff
@@ -509,14 +513,94 @@ Response:
...
+equal to the `MAX_CHECKOUT_LINE_QUANTITY` value.
+:::
+
+### Getting the available quantity for the particular address
+
+The quantity available can be fetched for the given address. The returned quantity
+is the sum of the quantity from all stocks that are available in the shipping zone that
+supports provided address. When the address is not provided the highest quantity
+from available shipping zones is returned.
```
Comment (pawelpikus) [inline docs/developer/products.mdx:531]:
```suggestion
supports provided address. When the address is not provided, the highest quantity
from available shipping zones is returned.
```

File: docs/developer/products.mdx [hunk 12]
```diff
@@ -509,14 +513,94 @@ Response:
...
+- In the case of a local collection point warehouse, the calculated quantity will equal
+  the quantity of this local warehouse.
+- In the case of a global collection point warehouse the quantity will equal the sum
+  of available quantity from all stocks that are available in the provided country code
+  and channel. When the country is not provided the quantity will be the sum of quantities
+  from all stocks that are available in the given channel, regardless of the shipping zone.
 
 ## Media
 
 Product media are images or videos that can be associated with products and used to render product galleries. The `Product` type contains two fields that refer to its media:
```
Comment (pawelpikus) [inline docs/developer/products.mdx:600]:
```suggestion
Product media are images or videos associated with products and used to render product galleries. The `Product` type contains two fields that refer to its media:
```

**Review by pawelpikus — CHANGES_REQUESTED** [review]
Great job on limiting the use of Passive voice. The guide is thus more human-readable and straightforward. I followed up with slight polishing, word choice, and minor grammar fixes.

File: docs/developer/products.mdx (hunk 11 above)
Comment (maarcingebala) [reply docs/developer/products.mdx:531]: What does "available" shipping zones mean? Does it mean "from all available shipping zones in the system"?

File: docs/developer/products.mdx [hunk 13]
```diff
@@ -509,14 +513,94 @@ Response:
...
+
+In the response, we get the available quantity of 20 as the `americas` is the only
+warehouse available in the `US`.
+
+In the case when the address is not provided, the quantity for the shipping zone with
+the highest available quantity is returned.
+
+### Handling collection point warehouses in the quantity calculations
+
+The collection point warehouses are treated as a magic single-warehouse shipping zone.
```
Comment (maarcingebala) [inline docs/developer/products.mdx:589]: If we say that something is "magic" I feel like we don't explain it properly, things should not be magic to our users :) 
```suggestion
The collection point warehouses are treated as a single-warehouse shipping zone.
```

File: docs/developer/products.mdx (hunk 11 above)
Comment (IKarbowiak) [reply docs/developer/products.mdx:531]: IMO, the `available shipping zones` are explained in the previous sentence

> The returned quantity is the sum of the quantity from all stocks that are available in the shipping zone that supports provided address.

**Pull request merged**